packages/rln-pcd/src/RLNPCD.ts

```diff
@@ -0,0 +1,229 @@
+import JSONBig from "json-bigint";
+import { v4 as uuid } from "uuid";
+import { Proof, RLN, RLNFullProof } from "rlnjs";
+import { Identity } from "@semaphore-protocol/identity";
+import { MerkleProof } from "@zk-kit/incremental-merkle-tree";
+
+import {
+  BigIntArgument,
+  ObjectArgument,
+  PCD,
+  PCDArgument,
+  PCDPackage,
+  SerializedPCD,
+  StringArgument,
+} from "@pcd/pcd-types";
+import {
+  SemaphoreIdentityPCD,
+  SemaphoreIdentityPCDPackage,
+} from "@pcd/semaphore-identity-pcd";
+import verificationKeyJSON from "../artifacts/16.json";
+
+let initArgs: RLNPCDInitArgs | undefined = undefined;
+
+export const RLNPCDTypeName = "rln-pcd";
+
+export interface RLNPCDInitArgs {
+  zkeyFilePath: string;
+  wasmFilePath: string;
+}
+
+// Ref: https://github.com/Rate-Limiting-Nullifier/rlnjs/blob/97fe15e04428c6adf81dbc856859e07527a063c9/src/types.ts#L59-L66
+export interface RLNPCDArgs {
+  // Identifier of the app. Every app using RLN should use a unique identifier.
+  rlnIdentifier: BigIntArgument;
+  // The semaphore keypair for a user
+  identity: PCDArgument<SemaphoreIdentityPCD>;
+  // The merkle proof of the user's identity commitment
+  merkleProof: ObjectArgument<MerkleProof>;
+  // The message that the user is sending
+  signal: StringArgument;
+  // The timestamp the message is sent
+  epoch: BigIntArgument;
+}
+
+// https://rate-limiting-nullifier.github.io/rln-docs/protocol_spec.html#technical-side-of-rln
+export interface RLNPCDClaim {
+  // The message that the user is sending
+  x: bigint;
+  // The timestamp the message is sent
+  epoch: bigint;
+  // Identifier of the app. Every app using RLN should use a unique identifier.
+  rlnIdentifier: bigint;
+  // The y value calculated from the polynomial of x
+  yShare: bigint;
+  // The merkle root of the identity commitment tree
+  merkleRoot: bigint;
+  // The unique identifier for (rlnIdentifier, epoch, identity)
+  internalNullifier: bigint;
+}
+
+export interface RLNPCDProof {
+  // snarkjs proof, including curve points and the protocol metadata
+  proof: Proof;
+  // The unique identifier for (rlnIdentifier, epoch)
+  externalNullifier: bigint;
+}
```

Comment:
> Since both of them are required for verifiers to interpret the proof, would it make sense for epoch and rlnIdentifier to be in the claim, and externalNullifier to be in the proof?

This makes sense to me. I think this division of fields between `claim` and `proof` looks good!